dask/dataframe/io/tests/test_parquet.py

```diff
@@ -3463,6 +3463,13 @@ def test_to_parquet_overwrite_raises(tmpdir, engine):
         dd.to_parquet(ddf, tmpdir, engine=engine, append=True, overwrite=True)
 
 
+def test_to_parquet_errors_non_string_column_names(tmpdir, engine):
+    df = pd.DataFrame({"x": range(10), 1: range(10)})
+    ddf = dd.from_pandas(df, npartitions=2)
+    with pytest.raises(ValueError, match="non-string column names"):
+        ddf.to_parquet(str(tmpdir.join("temp")), engine=engine)
```

Comment:
Fully optional suggestion that should not block merging. Since `to_parquet` understands `pathlib.Path` objects we could use the `tmp_path` pytest fixture and pass that directly to `to_parquet`

```suggestion
def test_to_parquet_errors_non_string_column_names(tmp_path, engine):
    df = pd.DataFrame({"x": range(10), 1: range(10)})
    ddf = dd.from_pandas(df, npartitions=2)
    with pytest.raises(ValueError, match="non-string column names"):
        ddf.to_parquet(tmp_path, engine=engine)
```